src/analysis/vector/geometry_checker/qgssinglegeometrycheck.h

```diff
@@ -0,0 +1,143 @@
+/***************************************************************************
+                      qgssinglegeometrycheck.h
+                     --------------------------------------
+Date                 : 6.9.2018
+Copyright            : (C) 2018 by Matthias Kuhn
+email                : matthias@opengis.ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSSINGLEGEOMETRYCHECK_H
+#define QGSSINGLEGEOMETRYCHECK_H
+
+#define SIP_NO_FILE
+
+#include <QList>
+#include <QCoreApplication>
+
+#include "qgsgeometry.h"
+#include "qgsgeometrycheck.h"
+
+#include "qgis_analysis.h"
+
+class QgsFeature;
+class QgsSingleGeometryCheck;
+
+/**
+ * An error from a QgsSingleGeometryCheck.
+ *
+ */
+class ANALYSIS_EXPORT QgsSingleGeometryCheckError
+{
+  public:
+    QgsSingleGeometryCheckError( const QgsSingleGeometryCheck *check, const QgsGeometry &geometry, const QgsPoint &errorLocation, const QgsVertexId &vertexId = QgsVertexId() )
+      : mCheck( check )
+      , mGeometry( geometry )
+      , mErrorLocation( errorLocation )
+      , mVertexId( vertexId )
+    {}
+
+    virtual ~QgsSingleGeometryCheckError() = default;
+
+    virtual void update( const QgsSingleGeometryCheckError *other );
```

Comment:
Dox

Reply:
Still trying to figure out what that actually does :D

Reply:
@m-kuhn When an error is fixed, affected geometries are re-scanned, and the new list of found errors is compared to the previously found errors. If a new error is equal (or close match) to an old one, the old error is updated (i.e. with new coordinates or indices, as a consequence of the changes performed).

Reply:
Thanks @manisandro do the docs look good to you?

Reply:
Yep looks good